packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts

```diff
@@ -0,0 +1,87 @@
+import { builtinModules } from "module";
+import { isAbsolute, relative } from "path";
+
+import type { Plugin } from "esbuild";
+
+import { RemixConfig } from "../../config";
+
+export function serverBareModulesPlugin(
+  remixConfig: RemixConfig,
+  dependencies: Record<string, string>,
+  onWarning?: (warning: string, key: string) => void
+): Plugin {
+  return {
+    name: "bare-modules",
+    setup(build) {
+      build.onResolve({ filter: /.*/ }, ({ importer, path }) => {
+        if (
+          !isBareModuleId(path) ||
+          path === "remix" ||
+          path === "@remix-run/server-entry" ||
+          path === "@remix-run/assets"
```

Comment:
Should be `@remix-run/assets-manifest`. Should we have this in a constant somewhere?

```
import { assetsManifestVirtualModuleId } from 'somewhere'
```